mne/tests/test_epochs.py

```diff
@@ -3680,3 +3682,192 @@ def test_add_channels_picks():
+        description=['x', 'y', 'z'],
+        orig_time=orig_date
+    )
+    raw.set_annotations(ants)
+    epochs = make_fixed_length_epochs(raw, duration=1, overlap=0.5)
+
+    # add Annotations to Epochs metadata
+    epochs.add_annotations_to_metadata()
+    metadata = epochs.metadata
+    assert 'Annotations_onset' in metadata.columns
+    assert 'Annotations_duration' in metadata.columns
+    assert 'Annotations_description' in metadata.columns
+
+    # check that the annotations themselves should be equivalent
+    # because first_samp offsetting occurs in Raw
+    assert_array_equal(
+        raw.annotations.onset,
+        epochs.annotations.onset
+    )
+    assert_array_equal(raw.annotations.duration,
+                       epochs.annotations.duration)
+    assert_array_equal(raw.annotations.description,
+                       epochs.annotations.description)
+
+    # compare Epoch annotations with expected values
+    epoch_ants = epochs.get_annotations_per_epoch()
+    if orig_date is None:
+        expected_annot_times = [
+            [],
+            [[.6, ant_dur, 'x'], [.7, ant_dur, 'y']],
+            [[.1, ant_dur, 'x'], [.2, ant_dur, 'y']],
+            [[.6, ant_dur, 'z']],
+            [[.1, ant_dur, 'z']],
+            [],
+            []
+        ]
+    else:
+        expected_annot_times = [
+            [],
+            [],
+            [],
+            [[.6, ant_dur, 'x'], [.7, ant_dur, 'y']],
+            [[.1, ant_dur, 'x'], [.2, ant_dur, 'y']],
+            [[.6, ant_dur, 'z']],
+            [[.1, ant_dur, 'z']],
+        ]
+    assert len(expected_annot_times) == len(epoch_ants)
+    for (x, y) in zip(epoch_ants, expected_annot_times):
+        if orig_date is not None:
+            # when orig_date is set + first_samp, those will offset
+            # the onset when Raw sets annotations. These should
+            # then be offset accordingly when Epochs look for annotations
+            assert_array_almost_equal([_x[0] for _x in x], [
+                _y[0] - raw._first_time for _y in y])
+        else:
+            # onset relative to Epoch start
+            assert_array_almost_equal([_x[0] for _x in x], [_y[0] for _y in y])
+
+        # duration
+        assert_array_equal([_x[1] for _x in x], [_y[1] for _y in y])
+
+        # description should be exactly the same
+        assert_array_equal([_x[2] for _x in x], [_y[2] for _y in y])
+
+
+def test_epoch_annotations_cases():
+    """Test Epoch Annotations different cases.
+
+    Here, we test the following cases crossed:
+    - annotation start is before/after epoch start
+    - annotation end is before/after epoch end
+    - 1 annotation that is fully outside all epochs (make sure it is dropped)
+    - 1 annotation that spans multiple epochs (make sure it shows up in both)
+
+    In addition, tests functionality when Epochs are loaded vs not.
+    """
+    # do a more complicated case
+    data = np.random.randn(1, 600) * 10e-12
```

Comment:
It's better practice to use a seeded RNG rather than whatever the current state is when pytest happens to call this (it makes the test more repeatable, even if the actual data values themselves do not matter).
```suggestion
    data = np.random.RandomState(0).randn(1, 600) * 10e-12
```